corehq/form_processor/tests/test_case_dbaccessor.py

```diff
@@ -0,0 +1,331 @@
+            name='doc',
+            content_type='text/xml'
+        ))
+        FormProcessorSQL.save_case(case)
+
+        with self.assertRaises(AttachmentNotFound):
+            CaseAccessorSQL.get_attachment_by_name(case.case_id, 'missing')
+
+        with self.assertNumQueries(1):
+            attachments = CaseAccessorSQL.get_attachments(case.case_id)
+
+        self.assertEqual(2, len(attachments))
+        sorted_attachments = sorted(attachments, key=lambda x: x.name)
+        for att in attachments:
+            self.assertEqual(case.case_id, att.case_id)
+        self.assertEqual('doc', sorted_attachments[0].name)
+        self.assertEqual('pic.jpg', sorted_attachments[1].name)
+
+    def test_get_transactions(self):
+        form_id = uuid.uuid4().hex
+        case = _create_case(form_id=form_id)
+        transactions = CaseAccessorSQL.get_transactions(case.case_id)
+        self.assertEqual(1, len(transactions))
+        self.assertEqual(form_id, transactions[0].form_id)
+
+        form_ids = _create_case_transactions(case)
+
+        transactions = CaseAccessorSQL.get_transactions(case.case_id)
+        self.assertEqual(4, len(transactions))
+        self.assertEqual([form_id] + form_ids, [t.form_id for t in transactions])
+
+    def test_get_transactions_for_case_rebuild(self):
+        form_id = uuid.uuid4().hex
+        case = _create_case(form_id=form_id)
+        form_ids = _create_case_transactions(case)
+
+        transactions = CaseAccessorSQL.get_transactions_for_case_rebuild(case.case_id)
+        self.assertEqual(2, len(transactions))
+        self.assertEqual([form_id, form_ids[0]], [t.form_id for t in transactions])
+
+    def test_get_case_by_location(self):
+        case = _create_case(case_type=SupplyPointCaseMixin.CASE_TYPE)
+        location_id = uuid.uuid4().hex
+        case.location_id = location_id
+        FormProcessorSQL.save_case(case)
+
+        fetched_case = CaseAccessorSQL.get_case_by_location(DOMAIN, location_id)
+        self.assertEqual(case.id, fetched_case.id)
+
+    def test_get_case_ids_in_domain(self):
+        case1 = _create_case(case_type='t1')
+        case2 = _create_case(case_type='t1')
+        case3 = _create_case(case_type='t2')
+
+        case_ids = CaseAccessorSQL.get_case_ids_in_domain(DOMAIN)
+        self.assertEqual({case1.case_id, case2.case_id, case3.case_id}, set(case_ids))
+
+        case_ids = CaseAccessorSQL.get_case_ids_in_domain(DOMAIN, 't1')
+        self.assertEqual({case1.case_id, case2.case_id}, set(case_ids))
+
+        case2.domain = 'new_domain'
+        FormProcessorSQL.save_case(case2)
+
+        case_ids = CaseAccessorSQL.get_case_ids_in_domain(DOMAIN)
+        self.assertEqual({case1.case_id, case3.case_id}, set(case_ids))
+
+
+def _create_case(domain=None, form_id=None, case_type=None):
+    """
+    Create the models directly so that these tests aren't dependent on any
+    other apps. Not testing form processing here anyway.
+    :return: case_id
```

Comment:
wow that's brilliant. I should do that more for stuff that doesn't actually test form processing.